assets/src/application/ui/layout/modal/ModalWithAlert.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useEffect } from 'react';
+import { useDisclosure } from '@chakra-ui/core';
+import { __ } from '@wordpress/i18n';
+import { canUseDOM } from '@appServices/utilities/dom';
+import { ESCAPE } from '@wordpress/keycodes';
+
+import { AlertDialog } from '@infraUI/display';
+import { Button as ButtonAdapter } from '@infraUI/inputs';
+import { Modal, ModalProps } from '@infraUI/layout/modal';
+
+interface Props extends ModalProps {
+	cancelBtnText?: string;
+	header?: string;
+	okBtnText?: string;
+	showAlertOnEscape: boolean;
+}
+
+const ModalWithAlert: React.FC<Props> = ({ children, showAlertOnEscape, ...props }) => {
+	const { isOpen, onOpen, onClose } = useDisclosure();
+	const cancelRef = React.useRef();
+	const cancelBtnText = props.cancelBtnText || __('No');
+	const header = props.header || __('Are you sure you want to close this?');
+	const okBtnText = props.okBtnText || __('Yes');
+	const onEscape = ({ keyCode }) => keyCode === ESCAPE && onOpen();
+
+	useEffect(() => {
+		canUseDOM && document.addEventListener('keydown', onEscape);
+
+		return () => {
+			canUseDOM && document.removeEventListener('keydown', onEscape);
+		};
```

Comment:
For clarity, better use `if (canUseDOM) {}`

Reply:
I followed the same example as in chakra: https://github.com/chakra-ui/chakra-ui/blob/master/packages/chakra-ui/src/Modal/index.js#L131
***
but if the clarity is an issue, then it's better to make it more explicit, thx for letting me know this :+1: